src/datanode/src/datanode.rs

```diff
@@ -402,7 +408,11 @@ impl Datanode {
 
     /// Start only the internal component of datanode.
     pub async fn start_instance(&mut self) -> Result<()> {
-        self.instance.start().await
+        let _ = self.instance.start().await;
+        if let Some(task) = &self.heartbeat_task {
+            task.start().await?;
+        }
+        Ok(())
```

Comment:
After second thought, I feel that it's better to keep the old way. Heartbeat carries datanode stat, which is used by meta to select datanode to place regions. If heartbeat is started before the catalog manager, it can't fully reflect the correct stat of the very datanode. Not good for load balancing.

On the other hand, the region alive keeper is started only after heartbeat is started. See the start method of heartbeat task. So I think it won't cause too much damage (that falsely close regions).